packages/manager/src/features/Lish/Weblish.tsx

```diff
@@ -56,24 +62,78 @@ export class Weblish extends React.Component<Props, State> {
   connect() {
     const { weblish_url, ws_protocols } = this.props;
 
-    this.socket = new WebSocket(weblish_url, ws_protocols);
+    /* When this.socket != origSocket, the socket from this connect()
+     * call has been closed and possibly replaced by a new socket. */
+    const origSocket = new WebSocket(weblish_url, ws_protocols);
+    this.socket = origSocket;
+
+    this.lastMessage = '';
+    this.setState({error: ''});
 
     this.socket.addEventListener('open', () => {
       if (!this.mounted) {
         return;
       }
-      this.setState({ renderingLish: true }, () => this.renderTerminal());
+      this.setState({ renderingLish: true }, () => this.renderTerminal(origSocket));
+    });
+
+    this.socket.addEventListener('close', (evt) => {
+      if (this.socket !== origSocket) {
+        return;
+      }
+      this.socket = null;
+      this.terminal?.dispose();
+      this.setState({ renderingLish: false });
+      if (!this.mounted) {
+        return;
+      }
```

Comment:
We can combine these early returns

```suggestion
      if (this.socket !== origSocket || !this.mounted) {
        return;
      }
      this.socket = null;
      this.terminal?.dispose();
      this.setState({ renderingLish: false });
```

Reply:
The intention was to do nothing when `this.socket !== origSocket`, but to run the cleanup whenever `this.socket === origSocket`. But, when `!this.mounted`, we don't want to proceed with the rest of the close handling. So, I believe the code is correct as is.

Reply:
Gotcha, can we add comment(s) explaining this?

Reply:
Comments added.